flows/oauth2Flows.md

```diff
@@ -27,4 +27,55 @@
     scope : string
 }
 ```
-- Flow diagram: https://app.code2flow.com/flowcharts/6475a4953f4c2482c82a4352
\ No newline at end of file
+- Flow diagram: https://app.code2flow.com/flowcharts/6475a4953f4c2482c82a4352
+
+
+## 2. Get oauth2 client
+- Path: "/appid-<appId>/recipe/oauth2/client/<clientId>"
+- Method: GET
+- Input Params:
+    - ```clientId=string```
+- Output:
+```
+{
+  status: "OK",
+  name: string,
+  clientInfo : 
+    {
+       clientId: string,
+       clientSecret: string,
+       scopes: string[],
+       timeCreated: number
+    }
+  redirectUris: string[],
+  timeUpdated: number,
+  authorizationCodes: [
+    {
+      tenantId: string,
+      sessionHandle: string,
+      scopes: string[],
+      accessType: string,
+      codeChallenge: string,
+      codeChallengeMethod: string,
+      redirectUri: string,
+      timeCreated: number,
+      timeExpires: number
+    }
+  ],
+  issuedTokens: [
+    {
+      tenantId: string,
+      sessionHandle: string,
+      userId: string,
+      scopes: string[],
+      accessTokenHash: string,
+      refreshTokenHash: string,
+      timeCreated: number,
+      timeAccessTokenExpires: number,
+      timeRefreshTokenExpires: number
+    }
+  ]
+}
+
+```
+- Flow diagram: https://app.code2flow.com/flowcharts/6475cc5b3f4c2482c82a736a
```

Comment:
- no need to check for length of client id
- no need to check for client id being uuid
- rename `getOauth2AuthorizationTokenByClientId` to `getOauth2TokensByClientId`
- Filter the authorizationCodes and issuedTokens to remove expired tokens:
   - for authorizationCodes, we check timeExpires value
   - for issuedTokens, if refresh token is present, you filter out based on value of timeRefreshTokenExpires. Else you filter out based on timeAccessTokenExpires.